app/models/searchgov_url.rb

```diff
@@ -144,6 +152,30 @@ def validate_document
     raise SearchgovUrlError.new('Noindex per HTML metadata') if document.noindex?
   end
 
+  def check_canonical_url
+    document.canonical_url.present? ? use_canonical_url : index_and_update_status
+  end
+
+  def use_canonical_url
+    canonical_url = document.canonical_url
+    return if canonical_url.nil?
+
+    check_for_redirection(canonical_url)
+    create_canonical_url(canonical_url)
+  end
+
+  def create_canonical_url(canonical_url)
+    searchgov_url = SearchgovUrl.create!(url: canonical_url)
```

Comment:
What if a `SearchgovUrl` already exists for this canonical url?  (Either because we've been through this fetch/redirection before, or because the canonical url appears elsewhere in the sitemap?).  If we attempt to create this `SearchgovUrl` here we're going to get a Validation error because of the duplicate url, right?

Reply:
Yes, we are gonna get a Validation error here and no duplicate `SearchgovUrl` is created here

Reply:
In that case, we will only ever see the `last_crawl_status` pointing to the canonical URL one the very first fetch of that original URL.  Each time reindexing happens (when triggered manually or ever 30(?) days) after that initial run the `last_crawl_status` will only report the (confusing/not helpful) generic validation error.